graphistry/umap_utils.py

```diff
@@ -589,10 +589,10 @@ def umap(
             index = res._nodes.index
             if res._node is None:
                 logger.debug("-Writing new node name")
+                res._nodes[config.IMPLICIT_NODE_ID] = range(len(res._nodes))
+
                 res = res.nodes(  # type: ignore
-                    res._nodes.reset_index(drop=True)
-                    .reset_index()
-                    .rename(columns={"index": config.IMPLICIT_NODE_ID}),
```

Comment:
Why not just call it `__index_umap__` or something that has low prob of collision? then we can keep it safe for cudf?